src/keymap/keymap.rs

```diff
@@ -35,10 +35,10 @@ impl FromStr for Keymap{
 impl Keymap {
 	pub fn from_str(context: &Context, s: &str) -> Result<Self, ()> {
 		unsafe{
-			let mut v = s.as_bytes().to_vec();
-			v.push('\0' as u8);
-
-			let cstring = CString::from_vec_unchecked(v);
+			let cstring = match CString::new(s.to_owned()){
```

Comment:
This can be simplified to `let cstring = CString::new(s.to_owned()).map_err(())?` :panda_face: 

And after I add proper errors with the `thiserror` crate the `map_err` can simply be removed :)